include/mapbox/vector_tile.hpp

```diff
@@ -58,64 +53,64 @@ class feature {
     const layer& layer_;
     optional<mapbox::geometry::identifier> id;
     GeomType type = GeomType::UNKNOWN;
-    using packed_iter_type = protozero::iterator_range<protozero::pbf_reader::const_uint32_iterator>;
-    packed_iter_type tags_iter;
-    packed_iter_type geometry_iter;
+    packed_iterator_type tags_iter;
+    packed_iterator_type geometry_iter;
 };
 
 class layer {
 public:
     layer(protozero::data_view const& layer_view);
 
     std::size_t featureCount() const { return features.size(); }
-    std::unique_ptr<const feature> getFeature(std::size_t) const;
-    std::string getName() const;
+    protozero::data_view const& getFeature(std::size_t) const;
+    std::string const& getName() const;
     std::uint32_t getExtent() const { return extent; }
     std::uint32_t getVersion() const { return version; }
 
 private:
     friend class feature;
 
     std::string name;
-    uint32_t version = 9;
-    uint32_t extent = 4096;
-    std::map<std::string, uint32_t> keysMap;
+    std::uint32_t version = 9;
+    std::uint32_t extent = 4096;
+    std::map<std::string, std::uint32_t> keysMap;
```

Comment:
`The map was so that getValue could avoid the linear search to convert a key string to an index.` at https://github.com/mapbox/mapbox-gl-native/pull/5183#issuecomment-222729581 indicates a linear search was to be avoided.